resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/SubmittedDependencyUploaderImpl.scala

```diff
@@ -76,29 +82,32 @@ private[spark] class SubmittedDependencyUploaderImpl(
     Utils.tryWithResource(new FileOutputStream(filesTgz)) { filesOutputStream =>
       CompressionUtils.writeTarGzipToStream(filesOutputStream, files.map(_.getAbsolutePath))
     }
-    // TODO provide credentials properly when the staging server monitors the Kubernetes API.
-    val kubernetesCredentialsString = OBJECT_MAPPER.writer()
-      .writeValueAsString(KubernetesCredentials(None, None, None, None))
-    val labelsAsString = OBJECT_MAPPER.writer().writeValueAsString(podLabels)
+    val clientKeyBase64 = clientKeyFile.map(f => BASE_64.encode(Files.toByteArray(f)))
+    val clientCertBase64 = clientCertFile.map(f => BASE_64.encode(Files.toByteArray(f)))
+    val oauthTokenBase64 = oauthToken.map(token => BASE_64.encode(token.getBytes(Charsets.UTF_8)))
 
-    val filesRequestBody = RequestBody.create(
-      okhttp3.MediaType.parse(MediaType.MULTIPART_FORM_DATA), filesTgz)
-
-    val kubernetesCredentialsBody = RequestBody.create(
-      okhttp3.MediaType.parse(MediaType.APPLICATION_JSON), kubernetesCredentialsString)
+    val ownerMonitoringCredentials = StagedResourcesOwnerMonitoringCredentials(
+        clientKeyDataBase64 = clientKeyBase64,
+        clientCertDataBase64 = clientCertBase64,
+        oauthTokenBase64 = oauthTokenBase64)
+    val stagedResourcesOwner = StagedResourcesOwner(
+      ownerNamespace = podNamespace,
```

Comment:
I think this is a tradeoff. In this model, using the resource staging server doesn't require a cluster administrator to provision any extra users to be able to read the API objects of interest, in exchange for an API that's a bit more complex requiring the user to provide their credentials to handle the cleanup.